Dockerfile

```diff
@@ -1,45 +1,87 @@
-FROM ruby:3.3.1
+ARG RUBY_VERSION=3.3.1
+ARG NODE_VERSION=20.10.0
 
-RUN apt-get update -qq && \
-    apt-get install -y build-essential libssl-dev libpq-dev less vim nano libsasl2-dev
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
-RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
-
-RUN mkdir -p /etc/apt/keyrings
-RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
-
-ENV NODE_MAJOR=18
-RUN echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list
+# Use Node image so we can pull the binaries from here.
+FROM node:$NODE_VERSION as node
 
-RUN apt update && apt install -y yarn nodejs
+# Ruby build image.
+FROM ruby:${RUBY_VERSION}-slim as base
 
+# Setup environment variables.
 ENV WORK_ROOT /src
-ENV APP_HOME $WORK_ROOT/myapp/
+ENV APP_HOME $WORK_ROOT/app
 ENV LANG C.UTF-8
-ENV GEM_HOME $WORK_ROOT/bundle
-ENV BUNDLE_BIN $GEM_HOME/gems/bin
-ENV PATH $GEM_HOME/bin:$BUNDLE_BIN:$PATH
+ENV BUNDLE_PATH $APP_HOME/vendor/bundle
+
+# Set prod environment to avoid installing dev dependencies
+ENV BUNDLE_WITHOUT development:test
+ENV BUNDLE_DEPLOYMENT 1
+ENV RAILS_ENV production
+ENV NODE_ENV production
```

Comment:
should we specify the yarn version as well?

Reply:
https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/Dockerfile.tt#L38

Reply:
hmm good point. I think we should either specify the version in the Dockerfile or not specify a version number at all.